packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_date.dart

```diff
@@ -74,7 +74,7 @@ class TemporalDate {
 
     // Validate
     String? regexString = regExp.stringMatch(iso8601String);
-    if (regexString != iso8601String) {
+    if (regexString == null || regexString != iso8601String) {
```

Comment:
@HuiSF we will be merging in null safety changes to the datastore plugin **interface** now but no changes to the datastore plugin. 

This is because the amplify flutter plugin has dependencies on all plugin interfaces but not the actual plugins themselves.